lib/solarwinds_apm/support/otlp_endpoint.rb

```diff
@@ -0,0 +1,196 @@
+
+module SolarWindsAPM
+  # OTLPEndPoint
+  class OTLPEndPoint
+    SW_ENDPOINT_REGEX = /^apm\.collector\.([a-z0-9-]+)\.cloud\.solarwinds\.com$/
+    OTEL_ENDPOINT_REGEX = %r{^https://otel\.collector\.([a-z0-9-]+)\.cloud\.solarwinds\.com:443(?:/.*)?$}
+    OTEL_ENDPOINT_LOCAL_REGEX = %r{\Ahttp://0\.0\.0\.0:(4317|4318)\z}
+    OTEL_ENDPOINT_LOCAL_REGEX2 = %r{\Ahttp://0\.0\.0\.0:(4317|4318)/v1/(metrics|traces|logs)\z}
+    DEFAULT_OTLP_ENDPOINT = 'https://otel.collector.na-01.cloud.solarwinds.com:443'
+    DEFAULT_APMPROTO_ENDPOINT = 'apm.collector.na-01.cloud.solarwinds.com'
+
+    def initialize
+      @token = nil
+      @service_name = nil
+      @lambda_env = determine_lambda_env
+      @agent_enable = true
+      @localhost = false
+      determine_if_localhost
+    end
+
+    def determine_if_localhost
+      @localhost = true if ENV['OTEL_EXPORTER_OTLP_ENDPOINT'].to_s.match?(OTEL_ENDPOINT_LOCAL_REGEX)
+      @localhost = true if ENV['OTEL_EXPORTER_OTLP_TRACES_ENDPOINT'].to_s.match?(OTEL_ENDPOINT_LOCAL_REGEX2)
+      @localhost = true if ENV['OTEL_EXPORTER_OTLP_METRICS_ENDPOINT'].to_s.match?(OTEL_ENDPOINT_LOCAL_REGEX2)
+      @localhost = true if ENV['OTEL_EXPORTER_OTLP_LOGS_ENDPOINT'].to_s.match?(OTEL_ENDPOINT_LOCAL_REGEX2)
+    end
+
+    def config_otlp_endpoint
+      config_service_name
+      config_token
+      %w[TRACES METRICS LOGS].each { |data_type| configure_otlp_endpoint(data_type) }
+    end
+
+    def config_service_name
+      resource_attributes = ENV['OTEL_RESOURCE_ATTRIBUTES'].to_s.split(',').each_with_object({}) do |resource, hash|
+        key, value = resource.split('=')
+        hash[key] = value
+      end
+
+      @service_name = if @lambda_env
+                        ENV['OTEL_SERVICE_NAME'] || ENV['AWS_LAMBDA_FUNCTION_NAME'] || resource_attributes['service.name']
+                      else
+                        ENV['OTEL_SERVICE_NAME'] || resource_attributes['service.name'] || @service_name || 'None'
+                      end
+
+      ENV['OTEL_SERVICE_NAME'] = @service_name
+    end
+
+    def mask_token(token)
+      token = token.to_s
+      return '*' * token.length if token.length <= 4
+
+      "#{token[0, 2]}#{'*' * (token.length - 4)}#{token[-2, 2]}"
+    end
+
+    def config_token(data_type)
+      agent_enable = true
+
+      if @localhost
+        # for localhost, only valid SW_APM_SERVICE_KEY can make agent enable
+        @agent_enable = false unless valid?(ENV['SW_APM_SERVICE_KEY'])
+        return
+      end
+
+      data_type_upper = data_type.upcase
+      data_type = data_type.downcase
+
+      if @lambda_env
+        # for case 10 and 11, lambda only care about SW_APM_API_TOKEN, not SW_APM_SERVICE_KEY
+        agent_enable = !ENV['SW_APM_API_TOKEN'].nil?
```

Comment:
it's actually the otelcol extension for lambda that care about this. instrumentation should ignore `SW_APM_API_TOKEN` and not require SW APM service key.

Reply:
removed the confusing comments

Reply:
Can we remove also the logic? inside lambda, the instrumentation layer doesn't care about either the SW APM service key or the API Token. It is enabled unless SW_APM_ENABLED=false aka the "SW APM disable setting".